crates/relayer/src/chain/requests.rs

```diff
@@ -134,6 +138,46 @@ impl From<PageRequest> for RawPageRequest {
     }
 }
 
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub enum Paginate {
+    #[default]
+    All,
+
+    PerPage {
+        pagination: u64,
```

Comment:
Is this field ever use?

Reply:
I will rename to:

```
    PerPage {
        limit: u64,
        max_queries: u64,
    },
```

Reply:
imho it should contain two fields: one which specifies how many results per page we want and one which specifies how many results total we want.

Reply:
So maybe `per_page: u64` and `total: u64`?

Reply:
then we can just stop issuing queries once we either a) have `total` results or b) there is no more next page/key

Reply:
Oh I had it set to one field for how many results per page and another one for the maximum number of queries. I will change to the option you suggested